winml/lib/Api.Experimental/LearningModelBuilder.cpp

```diff
@@ -0,0 +1,79 @@
+﻿#include "pch.h"
+#include "LearningModelBuilder.h"
+#include "LearningModel.h"
+#include "TensorFeatureDescriptor.h"
+#include "LearningModelSession.h"
+#include "LearningModelInputs.h"
+#include "LearningModelOutputs.h"
+#include "LearningModelOperatorSet.h"
+#include "OnnxruntimeProvider.h"
+
+namespace WINML_EXPERIMENTALP {
+
+LearningModelBuilder::LearningModelBuilder(int64_t opset) : inputs_(nullptr), outputs_(nullptr), operators_(nullptr), inert_session_(nullptr) {
+  WINML_THROW_IF_FAILED(CreateOnnxruntimeEngineFactory(engine_factory_.put()));
+  WINML_THROW_IF_FAILED(engine_factory_->CreateEmptyModel(opset, model_.put()));
+  inputs_ = winrt::make<winml_experimentalp::LearningModelInputs>(*this);
+  outputs_ = winrt::make<winml_experimentalp::LearningModelOutputs>(*this);
+  operators_ = winrt::make<winml_experimentalp::LearningModelOperatorSet>(*this);
+
+  winrt::com_ptr<_winml::IEngineBuilder> builder;
+  WINML_THROW_IF_FAILED(engine_factory_->CreateEngineBuilder(builder.put()));
+  winrt::com_ptr<_winml::IEngine> engine;
+  WINML_THROW_IF_FAILED(builder->CreateEngine(engine.put()));
```

Comment:
Would this call initialize OrtSession?

Reply:
That would mean Onnxruntime.dll would be loaded during LearningModel building right?

Reply:
Yes. This is strictly not needed for the builder, but creation of MLValues requires IEngine objects that are backed by OrtSessions. This can probably be replaced with a method "CreateInertEngine" or a InertEngine implementation of IEngine to avoid its creation. However, for now it seemed like there was little impact of just initializing a dummy OrtSession and passing it through.